src/app/InteractionModelDelegate.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the delegate SDK consumer needs to implement, which can talk to interaction model.
+ *
+ */
+
+#pragma once
+
+#include <core/CHIPCore.h>
+#include <messaging/ExchangeContext.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace app {
+class ReadClient;
+struct EventPathParams;
+
+/**
+ * @brief
+ *   This class defines the API for a delegate that an SDK consumer can use to interface with the interaction model.
+ */
+class InteractionModelDelegate
+{
+public:
+    /**
+     * Notification that the interaction model has received a list of events in response to a Read request and that list
+     * of events needs to be processed.
+     * @param[in]  apExchangeContext   An exchange context that represents the exchange the Report Data came in on.
```

Comment:
```suggestion
     * @param[in]  apExchangeContext   An exchange context that represents the exchange the Report Data came in on.
     *                                 This can be used to recover the NodeId of the node that sent the Report Data.
```

Reply:
Done, thanks